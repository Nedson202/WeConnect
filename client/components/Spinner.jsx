import React from 'react';

/**
 * @description react stateless component for card template
 * 
 * @return {Object} action dispatched by the action creator
 */
const Spinner = () => {
  return (
    <div className="spinner">
      <i className="fa fa-circle-o-notch fa-spin" />
    </div>
  );
}

export default Spinner;