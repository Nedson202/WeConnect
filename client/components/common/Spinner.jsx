import React from 'react';
import PropTypes from 'prop-types';
import { RingLoader } from 'react-spinners';

/**
 * @description react stateless component for card template
 *
 * @return {Object} spinner
 */
const Spinner = ({ isLoading }) => (
  <div className="spinner">
    <RingLoader
      color="#1a3c5d"
      loading={isLoading}
    />
  </div>
);

Spinner.propTypes = {
  isLoading: PropTypes.bool
};

Spinner.defaultProps = {
  isLoading: true
};

export default Spinner;
