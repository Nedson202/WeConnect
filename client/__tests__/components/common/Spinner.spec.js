import React from 'react';
import { configure, shallow, mount } from 'enzyme';
import thunk from 'redux-thunk';
import Adapter from 'enzyme-adapter-react-16';
import toJSON from 'enzyme-to-json';
import Spinner from '../../../components/common/Spinner';

configure({ adapter: new Adapter() });

let props;

const setup = () => {
  props = {
    isLoading: false,
  }

  return shallow(<Spinner {...props} />);
};

describe('Component: Spinner', () => {
  it('should render', () => {
    const wrapper = setup();
    expect(wrapper.find('div').length).toBe(1);
    expect(toJSON(wrapper)).toMatchSnapshot();
    expect(wrapper.length).toBe(1);
  }) 
});