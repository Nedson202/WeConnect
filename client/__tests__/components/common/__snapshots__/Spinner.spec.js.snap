// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Spinner should render 1`] = `
<div
  className="spinner"
>
  <onlyUpdateForKeys(Loader)
    color="#1a3c5d"
    loaderStyle={Object {}}
    loading={false}
    size={60}
    sizeUnit="px"
  />
</div>
`;
